### PR TITLE
Updating datastore to use v1

### LINKED ISSUE #2105: Update gcloud-python/datastore to use the v1 API
The v1 API is now available.

The protos are identical (just a different package name). The only behavioral change is captured in the release notes:
https://cloud.google.com/datastore/release-notes

### PR DESCRIPTION
**Pull request by dhermes**
Fixes #2105

Used `Makefile.datastore` (after updating it) to re-generate the generated modules

### CODE REVIEW

File: gcloud/datastore/_generated/_datastore.proto [hunk 1]
```diff
@@ -258,13 +257,33 @@ message Mutation {
...
   }
+
+  // When set, the server will detect whether or not this mutation conflicts
+  // with the current version of the entity on the server. Conflicting mutations
+  // are not applied, and are marked as such in MutationResult.
+  oneof conflict_detection_strategy {
+    // The version of the entity that this mutation is being applied to. If this
+    // does not match the current version on the server, the mutation conflicts.
+    int64 base_version = 8;
+  }
```
Comment (tseaver) [inline gcloud/datastore/_generated/_datastore.proto]: Are we exposing this feature anywhere?

File: gcloud/datastore/_generated/_query.proto [hunk 2]
```diff
@@ -279,4 +293,14 @@ message QueryResultBatch {
...
+
+  // The version number of the snapshot this batch was returned from.
+  // This applies to the range of results from the query's `start_cursor` (or
+  // the beginning of the query if no cursor was given) to this batch's
+  // `end_cursor` (not the query's `end_cursor`).
+  //
+  // In a single transaction, subsequent query result batches for the same query
+  // can have a greater snapshot version number. Each batch's snapshot version
+  // is valid for all preceding batches.
+  int64 snapshot_version = 7;
```
Comment (tseaver) [inline gcloud/datastore/_generated/_query.proto]: Are we exposing this feature?

**Comment by tseaver** [comment]
LGTM.  Should we open issues for the two new features in the `v1` API?

File: gcloud/datastore/_generated/_query.proto [hunk 3]
```diff
@@ -50,6 +50,17 @@ message EntityResult {
...
+  // The version of the entity, a strictly positive number that monotonically
+  // increases with changes to the entity.
+  //
+  // This field is set for [`FULL`][google.datastore.v1.EntityResult.ResultType.FULL] entity
+  // results.
+  //
+  // For [missing][google.datastore.v1.LookupResponse.missing] entities in `LookupResponse`, this
+  // is the version of the snapshot that was used to look up the entity, and it
+  // is always set except for eventually consistent reads.
+  int64 version = 4;
```
Comment (tseaver) [inline gcloud/datastore/_generated/_query.proto]: Are we exposing this feature?

**Comment by tseaver** [comment]
#2112 tracks the new versioning / conflict detection feature of the V1 API.

File: gcloud/datastore/_generated/_datastore.proto (hunk 1 above)
Comment (dhermes) [inline gcloud/datastore/_generated/_datastore.proto]: Nope. We should file an issue?

File: gcloud/datastore/_generated/_query.proto (hunk 2 above)
Comment (dhermes) [inline gcloud/datastore/_generated/_query.proto]: Nope. Issue? Thanks for looking, I just assumed the changes were cosmetic.

**Comment by dhermes** [comment]
Jinx. D'oh.

**Pull request merged**